src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -124,9 +136,18 @@ private function doProcessValue($value, $isRoot = false)
      *
      * @return array
      */
-    private function autowireCalls(\ReflectionClass $reflectionClass, array $methodCalls)
+    private function autowireCalls(\ReflectionClass $reflectionClass, array $methodCalls, bool $isRoot): array
     {
-        foreach ($methodCalls as $i => $call) {
+        $this->methodCalls = $methodCalls;
```

Comment:
That looks weird to have to use a property to pass some information to another method. Can't `$methodCalls` just be an argument of `autowireMethod` ?

Reply:
I was hesitating, but if I pass it to `autowireMethod`, it must be by reference. I don't want to fix states and references (both should be avoided when possible, mixing them is worst...), so I chosen this way.

Reply:
Then maybe make is very explicit, and remove the `methodCalls` from the method signature? (only use `$this->methodCalls`, that have to be set before calling `autowireCalls`)

Reply:
what about passing a $state stdClass object instead of using properties?

Reply:
It will not be consistent with the rest of this class that already use properties.

Reply:
@sroze thanks for the hint, fixed